test/python/visualization/pulse_v2/test_generators.py

```diff
@@ -18,45 +18,119 @@
 
 from qiskit import pulse
 from qiskit.test import QiskitTestCase
-from qiskit.visualization.pulse_v2 import (drawing_objects,
-                                           generators,
-                                           types)
-from qiskit.visualization.pulse_v2.style import stylesheet
+from qiskit.visualization.pulse_v2 import drawing_objects, types, stylesheet, device_info
+from qiskit.visualization.pulse_v2.generators import (barrier,
+                                                      chart,
+                                                      frame,
+                                                      snapshot,
+                                                      waveform)
 
 
-class TestGenerators(QiskitTestCase):
-    """Tests for generators."""
+def create_instruction(inst, phase, freq, t0, dt):
+    """A helper function to create InstructionTuple."""
+    frame_info = types.PhaseFreqTuple(phase=phase, freq=freq)
+    return types.PulseInstruction(t0=t0, dt=dt, frame=frame_info, inst=inst)
 
-    def setUp(self) -> None:
-        self.style = stylesheet.QiskitPulseStyle()
 
-    @staticmethod
-    def create_instruction(inst, phase, freq, t0, dt):
-        """A helper function to create InstructionTuple."""
-        frame = types.PhaseFreqTuple(phase=phase, freq=freq)
-        return types.InstructionTuple(t0=t0, dt=dt, frame=frame, inst=inst)
+class TestWaveformGenerators(QiskitTestCase):
+    """Tests for waveform generators."""
+
+    def setUp(self) -> None:
+        style = stylesheet.QiskitPulseStyle()
+        self.formatter = style.formatter
+        self.device = device_info.OpenPulseBackendInfo(
+            name='test',
+            dt=1,
+            channel_frequency_map={
+                pulse.DriveChannel(0): 5.0e9,
+                pulse.DriveChannel(1): 5.1e9,
+                pulse.MeasureChannel(0): 7.0e9,
+                pulse.MeasureChannel(1): 7.1e9,
+                pulse.ControlChannel(0): 5.0e9,
+                pulse.ControlChannel(1): 5.1e9
+            },
+            qubit_channel_map={
+                0: [pulse.DriveChannel(0),
+                    pulse.MeasureChannel(0),
+                    pulse.AcquireChannel(0),
+                    pulse.ControlChannel(0)],
+                1: [pulse.DriveChannel(1),
+                    pulse.MeasureChannel(1),
+                    pulse.AcquireChannel(1),
+                    pulse.ControlChannel(1)]
+            })
+
+    def test_consecutive_index_all_equal(self):
+        """Test for helper function to find consecutive index with identical numbers."""
+        vec = np.array([1, 1, 1, 1, 1, 1])
+        ref_inds = np.array([True, False, False, False, False, True], dtype=bool)
+
+        inds = waveform._find_consecutive_index(vec, resolution=1e-6)
```

Comment:
We should avoid testing private methods.

Reply:
This logic is relatively complicated and really critical for visualization. Is there any proper way of testing this? This doesn't need to be exposed.